TombEngine/Objects/TR2/Entity/tr2_worker_dualrevolver.cpp

```diff
@@ -223,9 +228,11 @@ namespace TEN::Entities::Creatures::TR2
 					extraTorsoRot.y = AI.angle;
 				}
 
-				if (!creature->Flags)
+				if (creature->Flags == 0 && item->Animation.FrameNumber == GetFrameIndex(item, 0))
```

Comment:
We have `TestAnimFrame` function for this purpose.

Reply:
oh sorry will change that

Reply:
Done.